src/screen/dashboard.rs

```diff
@@ -1750,65 +1733,59 @@ impl Dashboard {
                 self.last_changed = Some(Instant::now());
 
                 if let Some((pane, _)) = result {
-                    return self.focus_pane(main_window, main_window.id, pane);
+                    return self.focus_pane(self.main_window(), pane);
                 }
             } else {
                 let (state, pane) = pane_grid::State::new(Pane::new(Buffer::Empty, config));
                 self.panes.main = state;
                 self.last_changed = Some(Instant::now());
-                return self.focus_pane(main_window, main_window.id, pane);
+                return self.focus_pane(self.main_window(), pane);
             }
         }
 
         Task::none()
     }
 
-    fn split_pane(
-        &mut self,
-        axis: pane_grid::Axis,
-        config: &Config,
-        main_window: &Window,
-    ) -> Task<Message> {
+    fn split_pane(&mut self, axis: pane_grid::Axis, config: &Config) -> Task<Message> {
         if let Some((window, pane)) = self.focus {
-            if window == main_window.id {
+            if window == self.main_window() {
                 let result = self
                     .panes
                     .main
                     .split(axis, pane, Pane::new(Buffer::Empty, config));
                 self.last_changed = Some(Instant::now());
                 if let Some((pane, _)) = result {
-                    return self.focus_pane(main_window, main_window.id, pane);
+                    return self.focus_pane(window, pane);
                 }
             }
         }
 
         Task::none()
     }
 
-    fn reset_pane(
-        &mut self,
-        main_window: &Window,
-        window: window::Id,
-        pane: pane_grid::Pane,
-    ) -> Task<Message> {
-        if let Some(state) = self.panes.get_mut(main_window.id, window, pane) {
+    fn reset_pane(&mut self, window: window::Id, pane: pane_grid::Pane) -> Task<Message> {
+        if let Some(state) = self.panes.get_mut(window, pane) {
             state.buffer.reset();
         }
 
         Task::none()
     }
 
-    fn close_pane(
-        &mut self,
-        main_window: &Window,
-        window: window::Id,
-        pane: pane_grid::Pane,
-    ) -> Task<Message> {
+    fn close_pane(&mut self, window: window::Id, pane: pane_grid::Pane) -> Task<Message> {
         self.last_changed = Some(Instant::now());
 
-        if window == main_window.id {
+        if window == self.main_window() {
+            self.focus_history = self
+                .focus_history
+                .clone()
+                .into_iter()
```

Comment:
`iter().filter().cloned()` would technically be more efficient as you only clone the retained values.